### PR TITLE
Edited scope parameter details in New-AzRoleAssignment.md

### PR DESCRIPTION
**Pull request by stefanpetter**
## Description

The Scope parameter shows as Required: False in the docs currently, but it is mandatory when running the command, so I adjusted that. I also cleared up the examples that were available.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Resources/Resources/help/New-AzRoleAssignment.md [hunk 1]
```diff
@@ -394,19 +394,17 @@ Accept wildcard characters: False
 ```
 
 ### -Scope
-The Scope of the role assignment.
-In the format of relative URI.
-For e.g.
-"/subscriptions/9004a9fd-d58e-48dc-aeb2-4a4aec58606f/resourceGroups/TestRG".
-If not specified, will create the role assignment at subscription level.
-If specified, it should start with "/subscriptions/{id}".
```
Comment (isra-fel) [inline src/Resources/Resources/help/New-AzRoleAssignment.md:402]: Hey Stephan, as the description, if Scope is not specified, it defaults to the current subscription (you can get by `Get-AzContext`, so it is indeed an optional parameter.

Comment (stefanpetter) [reply src/Resources/Resources/help/New-AzRoleAssignment.md:402]: Hi @isra-fel , thank for responding. I tried assigning a new role to a service principal without specifying the -Scope option, but it did not work. The parameter is mandatory, hence my change in the documentation. 
image

Comment (isra-fel) [reply src/Resources/Resources/help/New-AzRoleAssignment.md:402]: In this case you typed "-ObjectId" and "-RoleDefinitionId" so PowerShell knows you are calling this specific parameter set [RoleIdWithScopeAndObjectIdParameterSet](https://github.com/Azure/azure-powershell/blob/57b2691fccafc6796f35b77851f87e9a892a1058/src/Resources/Resources/help/New-AzRoleAssignment.md#roleidwithscopeandobjectidparameterset) , and `Scope` is required in it.
Some other parameter sets, like the [default one](https://github.com/Azure/azure-powershell/blob/57b2691fccafc6796f35b77851f87e9a892a1058/src/Resources/Resources/help/New-AzRoleAssignment.md#emptyparameterset-default), do not require `Scope` (notice the brackets around it), so overall it is optional. Hope I answered your question :)

Comment (stefanpetter) [reply src/Resources/Resources/help/New-AzRoleAssignment.md:402]: Okay. Thanks for the explanations :)

**Pull request closed without merging**